### PR TITLE
improve dev builds

### PR DESCRIPTION
**Pull request by therealprof**
This PR basically reverts #141 and plays with inline markers for some large dev build binary size gains

### CODE REVIEW

**Comment by burrbull** [comment]
@ryankurte can you review this PR?

**Comment by burrbull** [comment]
failed with
```
error: literal out of range for i32
```
on several targets.

File: src/generate/peripheral.rs [hunk 1]
```diff
@@ -48,14 +48,15 @@ pub fn render(
...
 
         unsafe impl Send for #name_pc {}
 
         impl #name_pc {
             /// Returns a pointer to the register block
-            pub fn ptr() -> *const #base::RegisterBlock {
-                #address as *const _
+            #[inline(always)]
+            pub const fn ptr() -> *const #base::RegisterBlock {
+                #address as _
```
Comment (burrbull) [inline src/generate/peripheral.rs]: incorrect cast here (`*i32` instead of `*<width>`)

Comment (therealprof) [reply src/generate/peripheral.rs]: That is very weird. Why could a elided type result in an incompatible pointer?

Comment (therealprof) [reply src/generate/peripheral.rs]: Most likely the issue is not this change but the `PhantomMarker` change above. Anyway let's back that out.

**Comment by therealprof** [comment]
@ryankurte I removed the problematic change. Please have another look.

**Pull request merged**